src/H5timer.c

```diff
@@ -193,17 +193,18 @@ H5_now_usec(void)
         struct timespec ts;
 
         HDclock_gettime(CLOCK_MONOTONIC, &ts);
-        now = (uint64_t)(ts.tv_sec * (1000 * 1000)) + (uint64_t)(ts.tv_nsec / 1000);
+        now = ((uint64_t)ts.tv_sec * ((uint64_t)1000 * (uint64_t)1000)) +
+              ((uint64_t)ts.tv_nsec / (uint64_t)1000);
```

Comment:
Those casts obscure the arithmetic. Why not `now = (uint64_t)ts.tv_sec * 1000 * 1000 + ts.tv_nsec / 1000;` ?
Both `*` and `+` should yield a result with precision as great as the highest-precision operand.

Reply:
I prefer to be explicit that everything is done in uint64_t, since the precision of the arithmetic is important here. This way you don't need to know (or look up) the peculiarities of C type conversion to understand what is going on. I can remove some of the parentheses if you think they're a problem.

Reply:
We can maybe pull out the uint64_t from the 1000*1000:

now = ((uint64_t)ts.tv_sec *((uint64_t)(1000 * 1000)) + ((uint64_t)ts.tv_nsec / (uint64_t)1000);

Does that work?

Reply:
> Those casts obscure the arithmetic. Why not `now = (uint64_t)ts.tv_sec * 1000 * 1000 + ts.tv_nsec / 1000;` ? Both `*` and `+` should yield a result with precision as great as the highest-precision operand.

I tend to also prefer being explicit about the conversion for those in the future that may or may not  understand type conversion well. Would it maybe help to have some intermediate uint64_t variables for the `((uint64_t)1000 * (uint64_t)1000)` and `((uint64_t)ts.tv_nsec / (uint64_t)1000)`? Seems that would make the arithmetic less obscure and still allow for some explicit casts, even if not necessary in some places.

Reply:
I would agree with Dave, i.e. just add the cast to the initial member that needs conversion. No need to hammer it down :)

Reply:
I think Dave's suggestion sacrifices clarity for the sake of brevity, which I do not think is a good idea. Others have write access to this PR, if someone else wants to make the change go ahead, but I don't want the commit to have my name on it.